src/stake/cli.ts

```diff
@@ -59,27 +39,25 @@ const createAction = (parent: Command, createCmd: Command, network: Network) =>
       return { yes }
     })()
   }
-
-  const vars = await xe.vars(network.blockchain.baseURL)
-
-  const encWallet = await readWallet(opts.wallet)
+  const storage = withFile(opts.wallet)
 
   const api = xeWithNetwork(network)
-  const onChainWallet = await api.walletWithNextNonce(encWallet.address)
+  const onChainWallet = await api.walletWithNextNonce(await storage.address())
 
-  // fallback 0 is just for typing - stakeType is checked at top of func, so it should never be used
+  const vars = await xe.vars(network.blockchain.baseURL)
```

Comment:
Ensure that if this fails for any reason, a message is displayed informing the user that staking is currently unavailable and to try back later